frontend/src/metabase-lib/lib/queries/utils/nested-card-query-table.ts

```diff
@@ -42,7 +41,10 @@ export function getNestedCardTable(
     });
   }
 
-  return null;
+  // `card__123` tables are technically accessible via `metabase.table("card__123")`,
+  // but the `fields` on the table may be incorrect.
+  // see (metabase#25141)
+  return metadata.table(sourceTableId);
```

Comment:
There's a race condition in parts of the QB such that we can't always rely on `cards`. This fallback is necessary in those scenarios. Fixing #25141 should fix this issue and should also enable us to remove some of this complex virtual table logic.